src/main/flight/mixer.c

```diff
@@ -269,83 +273,121 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
+        crashflipRateAttenuator = gyroRateChange > smallAttChange ? 1.0f : gyroRateChange * smallAttChangeInv;
+
+        crashflipPower *= crashflipAttitudeAttenuator * crashflipRateAttenuator;
+    }
 
-        if (cosPhi < cosThreshold) {
-            // Enforce either roll or pitch exclusively, if not on diagonal
-            if (stickDeflectionRollAbs > stickDeflectionPitchAbs) {
-                signPitch = 0;
+    for (int i = 0; i < mixerRuntime.motorCount; ++i) {
+        float motorOutputNormalised =
+            signPitch * mixerRuntime.currentMixer[i].pitch +
+            signRoll * mixerRuntime.currentMixer[i].roll +
+            signYaw * mixerRuntime.currentMixer[i].yaw;
+
+        if (motorOutputNormalised < 0) {
+            if (mixerConfig()->crashflip_motor_percent > 0) {
+                motorOutputNormalised = -motorOutputNormalised * (float)mixerConfig()->crashflip_motor_percent / 100.0f;
             } else {
-                signRoll = 0;
+                motorOutputNormalised = 0;
             }
         }
 
-        // Apply a reasonable amount of stick deadband
-        const float crashFlipStickMinExpo = flipPowerFactor * CRASH_FLIP_STICK_MINF + power3(CRASH_FLIP_STICK_MINF) * (1 - flipPowerFactor);
-        const float flipStickRange = 1.0f - crashFlipStickMinExpo;
-        const float flipPower = MAX(0.0f, stickDeflectionExpoLength - crashFlipStickMinExpo) / flipStickRange;
-
-        for (int i = 0; i < mixerRuntime.motorCount; ++i) {
-            float motorOutputNormalised =
-                signPitch * mixerRuntime.currentMixer[i].pitch +
-                signRoll * mixerRuntime.currentMixer[i].roll +
-                signYaw * mixerRuntime.currentMixer[i].yaw;
-
-            if (motorOutputNormalised < 0) {
-                if (mixerConfig()->crashflip_motor_percent > 0) {
-                    motorOutputNormalised = -motorOutputNormalised * (float)mixerConfig()->crashflip_motor_percent / 100.0f;
-                } else {
-                    motorOutputNormalised = 0;
-                }
-            }
-            motorOutputNormalised = MIN(1.0f, flipPower * motorOutputNormalised);
-            float motorOutput = motorOutputMin + motorOutputNormalised * motorOutputRange;
+        motorOutputNormalised = MIN(1.0f, crashflipPower * motorOutputNormalised);
+        float motorOutput = motorOutputMin + motorOutputNormalised * motorOutputRange;
 
-            // Add a little bit to the motorOutputMin so props aren't spinning when sticks are centered
-            motorOutput = (motorOutput < motorOutputMin + CRASH_FLIP_DEADBAND) ? mixerRuntime.disarmMotorOutput : (motorOutput - CRASH_FLIP_DEADBAND);
+        // Force motors to disarm value unless motor signal exceeds crashflip motor deadband
+        motorOutput = (motorOutput < motorOutputMin + CRASHFLIP_MOTOR_DEADBAND) ? mixerRuntime.disarmMotorOutput : (motorOutput - CRASHFLIP_MOTOR_DEADBAND);
```

Comment:
Maybe CRASHFLIP_MOTOR_DEADBAND can be tested on motorOutputNormalised ? And make it 0...1 range (% of max power) ?

Reply:
this I truly don't know.  this part of the original code was a bit over my head.

Reply:
I can see that the current method costs us a tiny bit of output power.  That's because, if the demand is a motorOutputNormalised of 1.0, the maximum we allow, then when the dead-band is applied, we lose the deadband amount off the top of the motor signal.  But this deadband is only 20 motor units, a trivial amount.
The real deadband comes from the stick deadband, which is 15% of the stick travel.
When inside that distance:
- crashflipPower will be zero, 
- motorOutputNormalised will be zero
- motorOutput will be set to motorOutputMin
- disarm value is sent to motors, since motorOutput can then only be less than motorOutputMin plus deadband

All this seems OK to me, if a bit messy.

If you had a direct proposal to achieve the same result I will try it, but I think this is sort of right as is?

Reply:
OK I think my previous comment wasn't entirely right.

If the sticks are held out, and the quad rotates, the crashflipPower won't abruptly go to zero, it will ease back as the quad turns.

And I think I appreciate the intent of your comment better now.  

I've changed the deadband to 0.02f (2%) and compare motorOutputNormalised to it.  Then we get full power on output.  There will be a step from off to on, but tiny.

Reply:
Tested that and it works just the same.  Code seems tidier.